components/playlist/browser/mime_util.cc

```diff
@@ -0,0 +1,130 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/playlist/browser/mime_util.h"
+
+#include <optional>
+#include <utility>
+
+#include "base/containers/flat_map.h"
+#include "base/files/file_path.h"
+#include "base/no_destructor.h"
+
+namespace playlist {
+namespace {
+
+// References
+// * List of mimetypes registered to IANA
+//   * Video:
+//   https://www.iana.org/assignments/media-types/media-types.xhtml#video
+//   * Audio:
+//   https://www.iana.org/assignments/media-types/media-types.xhtml#audio
+// * Chromium media framework supports
+//   * media/base/mime_util_internal.cc
+// * Mimetype to extension
+//   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+//
+constexpr std::pair<std::string_view, base::FilePath::StringPieceType>
```

Comment:
1. use consexpr FixedFlatMap for mime->fileext map.
2. use flat_map for fileext->mime map.
3. use map.find() instead of contains+at.

Reply:
This might not be exactly same what you meant, but updated to use fixed_flat_map for mime=>ext map. In order to pick up the first mimetype when building ext=>mimemap, leaves the array of pairs.

Reply:
> ah, you're right, I haven't figure out the order that should be preserved to pick the right mimetype (I should've read the comment carefully).

Nah, I updated the comment in the latest commit. 

> Another take: maybe it's easier to just create two fixed flat maps with a clear usecase for each and have a unit test that makes sure both maps are in sync?

Yeah, what I was interested in was keeping maps in sync, so that makes sense. Plus, we can construct both maps at compile time 👍